packages/pipeline/src/entities/token_order.ts

```diff
@@ -0,0 +1,27 @@
+import { BigNumber } from '@0x/utils';
+import { Column, Entity, PrimaryColumn } from 'typeorm';
+
+import { OrderType } from '../types';
+import { bigNumberTransformer } from '../utils';
+
+@Entity({ name: 'token_orderbook_snapshots', schema: 'raw' })
+export class TokenOrderbookSnapshot {
+    @PrimaryColumn({ name: 'source' })
+    public source!: string;
+    @PrimaryColumn({ name: 'retrieval_timestamp', type: 'numeric', transformer: bigNumberTransformer })
+    public retrievalTimestamp!: BigNumber;
+    @PrimaryColumn({ name: 'order_type' })
+    public orderType!: OrderType;
+    @PrimaryColumn({ name: 'base_asset_symbol' })
+    public baseAssetSymbol!: string;
+    @Column({ name: 'base_asset_address' })
+    public baseAssetAddress!: string;
+    @PrimaryColumn({ name: 'base_volume' })
+    public baseVolume!: number;
```

Comment:
On the other hand, for volumes we should be using the `BigNumber` and `numeric` types. We don't have any hard guarantees about these numbers and in theory they could be very large or have a high number of significant digits. We want to make absolutely sure that we don't run into any precision-related errors here.